### PR TITLE
Reorganize package.json

### PR DESCRIPTION
**Pull request by rtravitz**
### Description

This PR supports [API-1322](https://vajira.max.gov/browse/API-1322). 

#### Notable changes
- Moves packages that are only used to build and test the application down to the `devDependencies` section of `package.json`. 
- Moves the `scripts` section to the top to match the default location after an `npm init`. 
- Moves babel configuration to its own file to make it easier to see at a glance which babel plugins are used when running a global search. The `package.json` will show up for plugins regardless of whether the package is used as part of the build or not.

#### What this PR does NOT do
- Add or remove any packages or scripts. Substantive changes will come in future PRs. This PR only exists to organize and take stock of which dependencies the app is currently using.

### Process

- [x] Tests added or updated for change
No changes are made to the functionality, and no tests need to be added or changed as a result.

- [x] Docs updated
There are no obvious documentation changes to make.

- [x] Accessibility concerns have been considered and addressed
There are no accessibility concerns with these changes.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -85,80 +59,106 @@
...
     "jest": "^25.5.4",
+    "jest-axe": "^3.2.0",
     "jest-environment-puppeteer": "^4.1.1",
     "jest-fetch-mock": "^3.0.3",
     "jest-image-snapshot": "^3.1.0",
     "jest-junit-reporter": "^1.1.0",
+    "jest-puppeteer": "^4.1.1",
+    "js-yaml-loader": "^1.2.2",
     "lodash.noop": "^3.0.1",
+    "markdown-component-loader": "github:adhocteam/markdown-component-loader#dv-3146-from110",
```
Comment (mjlumetta) [inline package.json:129]: we should uhh figure out why this is and try to do it in a sustainable way. not sure if that fits in this ticket or not, but it shouldn't hold up this PR.

Comment (rtravitz) [reply package.json:129]: Oh, there's a ticket for it 😏 @nfasulo

Comment (nfasulo) [reply package.json:129]: :fork_and_knife: I'm not seeing a corresponding issue in Jira but here is the original ticket for this kind of cleanup https://github.com/department-of-veterans-affairs/vets-contrib/issues/1277.

I found PR Drew created to update the original package: https://github.com/ticky/markdown-component-loader/pull/179/files

Comment (rtravitz) [reply package.json:129]: Forked libraries is the one I meant, of course!

**Pull request merged**